src/presentation/controllers/manufacturer.ts

```diff
@@ -0,0 +1,25 @@
+import { Request } from "express";
+import { IAddManufacturerUseCase } from "../../application/usecases/manufacturer";
+import { Controller, HttpResponse } from "../../domain/controller";
+import { HttpStatusCode } from "../../domain/http";
+import { Manufacturer } from "../../domain/manufacturer";
+
+export class AddManufacturerController implements Controller {
+  constructor(readonly addManufacturerUseCase: IAddManufacturerUseCase) {}
+  async handle(request: Request): Promise<HttpResponse> {
+    try {
+      const data = request.body as Manufacturer;
```

Comment:
talvez algo como:

```suggestion
      const data = request.body as ManufacturerRequest;
```

o que acha?

Reply:
ou ManufacturerBody

Reply:
Acredito que essa alteração não seja vantajosa, pois estamos utilizando esse tipo em várias partes do projeto, inclusive como tipo de retorno em diversos métodos. Implementar essa mudança pode aumentar a complexidade do sistema sem uma necessidade real. Por exemplo, atualmente usamos `const data = request.body as Manufacturer`, e a proposta é mudar para `const data = request.body as ManufacturerRequest`; essa alteração só faria sentido se fosse somente utilizado no controller.